tests/integration/backward_compatible/heartbeat_test.py

```diff
@@ -48,25 +53,55 @@ def collector(c, *_):
             connection_added_collector, connection_removed_collector
         )
 
-        self.simulate_heartbeat_lost(self.client, addr, 2)
+        assertion_succeeded = [False]
```

Comment:
Instead of making `assertion_succeeded` a list, we can use `nonlocal assertion_succeeded` in `assert_heartbeat_stopped_and_restored` to mutate its value.